deepspeed/runtime/zero/stage3.py

```diff
@@ -269,7 +269,7 @@ def __init__(self,
         self.__params_in_ipg_bucket: List[Parameter] = []
         self.is_gradient_accumulation_boundary: bool = True
 
-        self.__param_reduce_events: Deque[Event] = collections.deque()
+        self.__param_reduce_events: Deque[get_accelerator().Event] = collections.deque()
```

Comment:
This would break if the accelerator doesn't have `Event` defined. Perhaps we need an `event_type()` method for `*_Accelerator` classes?

Reply:
This should be get_accelerator().Event(), and the method call should return the event type.   I'll fix this.

https://github.com/microsoft/DeepSpeed/blob/master/accelerator/cuda_accelerator.py#L106

Reply:
Event and Stream is now define and impelmented as @abstractmethod + @property, so reference their names in data types would return torch.cuda.Event and torch.cuda.Stream.  We can also use the same interface to create instances of Event and Stream by add parenthsis after it:
https://github.com/microsoft/DeepSpeed/pull/2677/files#diff-063566e7665f62d56f954817035fb24b21a1395225ca5d1ae792e60220c98e42R88

`get_accelerator().Event`: would return torch.cuda.Event as class type
`get_accelerator().Event()`: would create new object from torch.cuda.Event

Reply:
Looks good, thank you!